### DISCUSSION TITLE
Use and return multiple hooks from a single call with records

### DISCUSSION THREAD

**Discussion by Kanyade** [rrousselGit/flutter_hooks#430] [Q&A]
I want to unify some common hooks into one, returning all of them at once utilizing records.
Would this have any side effects / drawbacks, can it be used at all like this? Or should I move to creating a custom class for it?

![image](https://github.com/rrousselGit/flutter_hooks/assets/55270082/422fa849-85e5-4aa4-aa4a-8c26be7bb315)

**Comment by rrousselGit** [comment]
You should make this a mixin, but rather a top-level function. Otherwise that's fine
